web/src/app/workplace/taskdetails/taskdetails.component.ts

```diff
@@ -1,80 +1,119 @@
-    private alertService: AlertService,
-    private errorModalService: ErrorModalService) {
+              private taskService: TaskService,
+              private workplaceService: WorkplaceService,
+              private router: Router,
+              private removeConfirmationService: RemoveConfirmationService,
+              private requestInProgressService: RequestInProgressService,
+              private alertService: AlertService,
+              private errorModalService: ErrorModalService) {
   }
 
   ngOnInit() {
+    this.currentWorkbasket = this.workplaceService.currentWorkbasket;
+    this.workbasketSubscription = this.workplaceService.getSelectedWorkbasket().subscribe(workbasket => {
+      this.currentWorkbasket = workbasket;
+    });
     this.routeSubscription = this.route.params.subscribe(params => {
       const id = params['id'];
       this.getTask(id);
     });
   }
 
   resetTask(): void {
-    this.task = { ...this.taskClone };
+    this.task = {...this.taskClone};
     this.task.customAttributes = this.taskClone.customAttributes.slice(0);
     this.task.callbackInfo = this.taskClone.callbackInfo.slice(0);
     this.alertService.triggerAlert(new AlertModel(AlertType.INFO, 'Reset edited fields'));
   }
 
   getTask(id: string): void {
     this.requestInProgress = true;
-    this.taskService.getTask(id).subscribe(task => {
+    if (id && id.indexOf('new-task') !== -1) {
       this.requestInProgress = false;
-      this.task = task;
-      this.cloneTask();
-      this.taskService.selectTask(task);
-    }, err => {
-      this.errorModalService.triggerError(
-        new ErrorModel('An error occurred while fetching the task', err));
-    });
+      this.task = new Task(undefined, new ObjectReference(), this.currentWorkbasket);
+    } else {
+      this.taskService.getTask(id).subscribe(task => {
+        this.requestInProgress = false;
+        this.task = task;
+        this.cloneTask();
+        this.taskService.selectTask(task);
+      }, err => {
+        this.errorModalService.triggerError(
+          new ErrorModel('An error occurred while fetching the task', err));
+      });
+    }
   }
 
-  updateTask() {
+  onSave() {
     this.requestInProgressService.setRequestInProgress(true);
+    this.task.taskId ? this.updateTask() : this.createTask();
+  }
+
+  updateTask() {
     this.taskService.updateTask(this.task).subscribe(task => {
       this.requestInProgressService.setRequestInProgress(false);
       this.task = task;
       this.cloneTask();
       this.taskService.publishUpdatedTask(task);
       this.alertService.triggerAlert(new AlertModel(AlertType.SUCCESS, 'Update successful!'))
-    }, err => {this.alertService.triggerAlert(new AlertModel(AlertType.DANGER, 'Update not successful!'))});
+    }, err => {
+      this.alertService.triggerAlert(new AlertModel(AlertType.DANGER, 'Update not successful!'))
+    });
+  }
+
+  createTask() {
+    this.addDateToTask();
```

Comment:
You are missing to activate general spinnner this.requestInProgressService.setRequestInProgress(true);

Reply:
I am doing that in onSave()